node/modules/actorevent.go

```diff
@@ -98,6 +98,43 @@ func EventFilterManager(cfg config.FevmConfig) func(helpers.MetricsCtx, repo.Loc
 	return func(mctx helpers.MetricsCtx, r repo.LockedRepo, lc fx.Lifecycle, cs *store.ChainStore, sm *stmgr.StateManager, evapi EventHelperAPI, chainapi full.ChainAPI) (*filter.EventFilterManager, error) {
 		ctx := helpers.LifecycleCtx(mctx, lc)
 
+		const (
+			// TODO: Consider introducing configuration parameters for constants below.
+			resolverCacheSize      = 1024
+			resolverCacheExpiry    = time.Minute
+			resolverCacheNilTipSet = false
+		)
+		actorResolver := filter.NewCachedActorResolver(
+			func(ctx context.Context, emitter address.Address, ts *types.TipSet) (abi.ActorID, error) {
+				var addr address.Address
+				if emitter.Protocol() == address.ID {
```

Comment:
Consider moving this short-circuit logic into the `StateManager` itself? Any caller of that method should benefit from it.